wcfsetup/install/files/js/WCF.Attachment.js

```diff
@@ -285,20 +292,112 @@ if (COMPILER_TARGET_DEFAULT) {
 		 * @see        WCF.Upload._upload()
 		 */
 		_upload: function (event, file, blob) {
-			var $uploadID = undefined;
+			var _super = this._super.bind(this);
 			
-			if (this._validateLimit()) {
-				$uploadID = this._super(event, file, blob);
-			}
-			
-			if (this._fileUpload) {
-				// remove and re-create the upload button since the 'files' property
-				// of the input field is readonly thus it can't be reset
-				this._removeButton();
-				this._createButton();
-			}
-			
-			return $uploadID;
+			require([
+				'WoltLabSuite/Core/FileUtil',
+				'WoltLabSuite/Core/ImageResizer',
+				'WoltLabSuite/Core/Ajax/Status'
+			]).then((function (modules) {
```

Comment:
Please use the regular syntax that we use pretty much everywhere: `require([…], function(…) {});`.

Reply:
Using the `Promise` returned by require allows nice error handling, consider the following

```
require(['WoltLabSuite/Core/Event/Handler'], function (x) { return Promise.reject(new Error('Callback')) }, function (err) { console.error('Error caught!', err) })

Promise {<resolved>: Arguments(1)}
VM217:1 Uncaught (in promise) Error: Callback
    at <anonymous>:1:83
    at WoltLabSuite.Core.min.js?v=1539810053:16

require(['WoltLabSuite/Core/Event/Handler']).then(function (x) { return Promise.reject(new Error('Callback')) }).catch(function (err) { console.error('Error caught!', err) })

VM223:1 Error caught! Error: Callback
    at <anonymous>:1:88
Promise {<resolved>: undefined}
```

Reply:
That's a bogus example, you're abusing the error handler for `require()` itself which also triggers if loading the module fails for whatever reasons. Remember, you're rejecting the promise of `require` and not of whatever you're doing inside.

```
require([…], function(…) {
  (new Promise()).catch(…);
});
```